src/autoscaler/db/sqldb/scheduler_sqldb_test.go

```diff
@@ -123,4 +124,83 @@ var _ = Describe("SchedulerSqldb", func() {
 
 	})
 
+	Describe("SyncActivesSchedules", func() {
+		BeforeEach(func() {
+			sdb, err = NewSchedulerSQLDB(url, logger)
+			Expect(err).NotTo(HaveOccurred())
+			err = cleanSchedulerActiveScheduleTable()
+			Expect(err).NotTo(HaveOccurred())
+
+			err = insertSchedulerActiveSchedule(111111, "app-id-1", 1, 2, 10, 5)
+			Expect(err).NotTo(HaveOccurred())
+			err = insertSchedulerActiveSchedule(222222, "app-id-2", 2, 3, 7, 5)
+			Expect(err).NotTo(HaveOccurred())
+			err = insertSchedulerActiveSchedule(333333, "app-id-3", 3, 5, 12, -1)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			err = sdb.Close()
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		JustBeforeEach(func() {
+			schedules, err = sdb.GetActiveSchedules()
+		})
+		Context("when appIdMap is empty", func() {
+			BeforeEach(func() {
+				appIdMap = map[string]bool{}
+				err = sdb.SynchronizeActiveSchedules(appIdMap)
```

Comment:
Why isn't this done in a higher JBE?
All you are doing in the BE is setting the appIdMap.